addons/mail/tests/test_mail_template.py

```diff
@@ -63,3 +65,31 @@ def test_mail_template_acl(self):
 
         with self.assertRaises(AccessError):
             mail_template.with_user(self.user_employee).name = 'Test write'
+
+    def test_mail_mail_schedule_date_after(self):
+        """Test that a mail scheduled in the future is not sent"""
+        # basic mail scheduled in two days
+        base_values = {
+            'body_html': '<p>Test</p>',
+            'email_to': 'test@example.com',
+            'scheduled_date': (datetime.now() + timedelta(days=1)).strftime(DEFAULT_SERVER_DATETIME_FORMAT)
```

Comment:
Just to be complete: we could also check that a void /False scheduled date is considered as "to be sent asap". We know it works but that way we cover all use cases.